### ISSUE TITLE
Weird extmark placement for first placeholder

### ISSUE THREAD

**Issue by bew** [L3MON4D3/LuaSnip#1020]
(this bug might be linked to #1019 (also with `cmp`))

With the following snippet:
```lua
snip("rbad", {desc = "bad local require"},
 { t"(manually) local ", i(1), t' = require"', i(2), t'"' }
)
```

![Peek 2023-09-28 00-38](https://github.com/L3MON4D3/LuaSnip/assets/9730330/c6c09213-f157-4501-b622-465a64665407)

Here I try to use the snippet 3 times:
1. with direct expansion via `ls.expand()` **BUG**
2. by first selecting something in `cmp` but still using a direct expansion via `ls.expand()` **ok**
3. by first selecting the snippet result then `<C-y>` **ok**

**Comment by L3MON4D3** [comment]
Weird!
I tried to reproduce it, and couldn't, at least with my current config.

https://github.com/L3MON4D3/LuaSnip/assets/41961280/d4a28a5f-d8eb-4a36-9c37-88a530720978

Here's my cmp-config, any major differences?
```lua
local cmp = require'cmp'
local types = require("cmp.types")
-- local session = require("session")

cmp.setup {
	completion = {
		autocomplete = {cmp.TriggerEvent.TextChanged},
		completeopt = "menu,menuone,select"
	},
	snippet = {
		expand = function(args)
			-- local override_snip = session.lsp_override_snips[args.body]
			-- if override_snip then
			-- 	require("luasnip").snip_expand(override_snip)
			-- else
				require("luasnip").lsp_expand(args.body)
			-- end
		end,
	},
	mapping = {
		['<C-d>'] = cmp.mapping.scroll_docs(-4),
		-- ['<C-f>'] = cmp.mapping.scroll_docs(4),
		['<C-y>'] = cmp.mapping.confirm(),
		['<C-n>'] = cmp.mapping.select_next_item({ behavior = types.cmp.SelectBehavior.Insert }),
		['<C-p>'] = cmp.mapping.select_prev_item({ behavior = types.cmp.SelectBehavior.Insert }),
	},
	sources = cmp.config.sources({
		{name = "path"},
		{name = "luasnip"}
	}),
	window = {
		documentation = false,
	},
	experimental = {
		native_menu = false,
		ghost_text = true
	},
}
```
Using cmp `5dce1b7`, luasnip master.

**Comment by bew** [comment]
Yeah that's what I thought, I'll try to repro with a minimal config later, probably this weekend.

**Comment by L3MON4D3** [comment]
Oh, one common cause of messed up extmarks are formatters, do you have one enabled?

**Comment by bew** [comment]
No no formatter at the moment

**Comment by bew** [comment]
It might be because I'm still using cmp's native menu.

Here is a reproduction script:
https://gist.github.com/bew/63bf7c3b01a32e85836e621a1b220e26
![Peek 2023-10-08 06-45](https://github.com/L3MON4D3/LuaSnip/assets/9730330/c8498fe8-7fa8-4baf-b631-b2326910bc0c)

**Comment by L3MON4D3** [comment]
Ah, nice!
Seems like others have also run into this issue, #721 for example.
I don't know the internals of cmp well, but there are relatively few `nvim_buf_set_text`/`nvim_buf_set_lines` IIRC, maybe you could track what text they modify?

**Comment by bew** [comment]
Note that I'm not using cmp to confirm & expand the snippet, I direclty use `<C-k>` to expand, so cmp's text changes shouldn't be happening
